network.tf

```diff
@@ -0,0 +1,91 @@
+locals {
+    public_cidr = ["10.0.0.0/24", "10.0.1.0/24"]
+    private_cidr = ["10.0.10.0/24", "10.0.11.0/24"]
+}
+
+# VPC
+resource "aws_vpc" "main" {
+  cidr_block = "10.0.0.0/16"
+}
+
+# subnet
+resource "aws_subnet" "public" {
+
+  count = 2
+
+  vpc_id     = aws_vpc.main.id
+  cidr_block = local.public_cidr[count.index]
+
+  tags = {
+    Name = "public${count.index}"
+  }
+}
+
+resource "aws_subnet" "private" {
+  count = 2
+
+  vpc_id     = aws_vpc.main.id
+  cidr_block = local.private_cidr[count.index]
+
+  tags = {
+    Name = "private${count.index}"
+  }
+}
+
+# IGW
+resource "aws_internet_gateway" "igw" {
+  vpc_id = aws_vpc.main.id
+
+  tags = {
+    Name = "main"
+  }
+}
+
+# EIP
+resource "aws_eip" "nat" {
+  count = 2
+
+  vpc      = true
+}
+
+# NGW
+resource "aws_nat_gateway" "ngw" {
+  count = 2 
+
+  allocation_id = aws_eip.nat[count.index].id
+  subnet_id     = aws_subnet.public[count.index].id
+
+  tags = {
+    Name = "main"
+  }
+  #depends_on = [aws_internet_gateway.example]
```

Comment:
It's completely up to your preference. However, having commented-out code is considered a bad styling

Reply:
Clear